tests/integration/test_uvicorn_logs.py

```diff
@@ -0,0 +1,93 @@
+# Copyright 2021 - 2023 Universität Tübingen, DKFZ, EMBL, and Universität zu Köln
+# for the German Human Genome-Phenome Archive (GHGA)
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+"""Isolated test for uvicorn that spins up the server and verifies the log output.
+
+There is a modified version of run_server that ensures the server shuts down gracefully
+when cancelling the task.
+"""
+import asyncio
+import json
+from contextlib import suppress
+from typing import Union
+
+import pytest
+import uvicorn
+from fastapi import FastAPI
+from hexkit.log import LoggingConfig, configure_logging
+
+from ghga_service_commons.api.api import ApiConfigBase, configure_app
+
+EXPECTED_FIELDS = {
+    "timestamp",
+    "service",
+    "instance",
+    "level",
+    "correlation_id",
+    "name",
+    "message",
+    "details",
+}
+
+
+async def run_server_with_shutdown(app: Union[FastAPI, str], config: ApiConfigBase):
+    """Copy of `run_server()` that allows server to shutdown if task is cancelled."""
+    uv_config = uvicorn.Config(
+        app=app,
+        host=config.host,
+        port=config.port,
+        log_config=None,
+        reload=config.auto_reload,
+        workers=config.workers,
+    )
+
+    server = uvicorn.Server(uv_config)
+    try:
+        await server.serve()
+    except asyncio.CancelledError:
+        await server.shutdown()
```

Comment:
Actually, this might be a good addition to the actual run_server function as well. Right, @Cito?

Reply:
I think you can just copy this over to the original run_server function, but let's wait for @Cito to have a look. Maybe I am missing something.

Reply:
@Cito, should we also be catching other/all error types so that we ensure the shutdown occurs gracefully in all cases and not just for a cancelled task?

Reply:
The main loop should not produce any other errors. But the startup could produce errors, and we don't want to shutdown in that case.

Reply:
Maybe even:

```python
    try:
        await server.serve()
    except asyncio.CancelledError:
        if server.started:
            await server.shutdown()
        raise
```

Reply:
@Cito & @KerstenBreuer please take a look at [f8c3f56](https://github.com/ghga-de/ghga-service-commons/pull/30/commits/f8c3f56fb6fd3b8c7df853b49d3eeb29eca87719)

Reply:
Looks good to me